UPGRADE.md

```diff
@@ -34,6 +39,126 @@ As part of the update of flysystem the support for the guzzle client package `gu
 
 The `GoogleGeolocator` and the `NominatimGeolocator` no longer support the Guzzle client and require a `Symfony\HttpClient` client instead.
 
+### Updating to flysystem 3
+
+The Sulu media storage uses now the Flysystem Bundle which need to be configured:
+
+```yaml
+# config/packages/flysystem.yaml
+flysystem:
+    storages:
+        default.storage:
+            adapter: 'local'
+            options:
+                directory: '%kernel.project_dir%/var/uploads'
+```
+
+Here are some examples on how to migrate individual providers:
+
+<details>
+  <summary>Sulu Media S3 to Flysystem S3</summary>
+
+Before:
+
+```yaml
+sulu_media:
+    storage: s3
+    storages:
+        s3:
+            key: 'your aws s3 key'
+            secret: 'your aws s3 secret'
+            bucket_name: 'your aws s3 bucket name'
+            path_prefix: 'optional/path/prefix'
+            region: 'eu-west-1'
+```
+
+New:
+
+```yaml
+flysystem:
+    storages:
+        default.storage:
+            adapter: 'aws'
+            options:
+                client: 'aws_client_service' # The service ID of the Aws\S3\S3Client instance
+                key: ''
+                secret: ''
+                bucket: 'bucket_name'
+                prefix: 'optional/path/prefix'
+                streamReads: true
+```
+
+</details>
+
+<details>
+  <summary>Sulu Google Cloud Config to Flysystem S3</summary>
+
+> [!NOTE]  
+> If you were using `superbalist/flysystem-google-storage` replace it with the official package version `league/flysystem-google-cloud-storage`.
+
+Before:
+
+```yaml
+sulu_media:
+    storage: google_cloud
+    storages:
+        google_cloud:
+            key_file_path: '/path/to/key.json'
+            bucket_name: 'sulu-bucket'
+            path_prefix: 'optional path prefix'
+```
+
+New:
+```yaml
+flysystem:
+    storages:
+        default.storage:
+            adapter: 'gcloud'
+            options:
+                client: 'gcloud_client_service' # The service ID of the Google\Cloud\Storage\StorageClient instance
```

Comment:
same as above how this service is created?

Reply:
Yes, if you're already using this. You should already have this service set up. You just need to reference it here.